packages/vite/src/node/utils.ts

```diff
@@ -189,8 +189,22 @@ const knownTsOutputRE = /\.(js|mjs|cjs|jsx)$/
 export const isTsRequest = (url: string) => knownTsRE.test(cleanUrl(url))
 export const isPossibleTsOutput = (url: string) =>
   knownTsOutputRE.test(cleanUrl(url))
-export const getTsSrcPath = (filename: string) =>
-  filename.replace(/\.([cm])?(js)(x?)(\?|$)/, '.$1ts$3')
+export const getPotentialTsSrcPaths = (filename: string) => {
+  const fileParts = filename.split(/(\.[cm]?js?\?|\.[cm]?js?$)/)
```

Comment:
Thanks for adding all the tests! Great to have them.

I have a question on this line:

Should the regex be `/(\.[cm]?js\?|\.[cm]?js$)/` (removed `?` after the `js`) to make the `s` compulsory?

I also wonder if we can shorten it as `/\.[cm]?js(\?|$)/`

Reply:
Good spot, I must've accidentally left it in when I removed the checks for `.jsx`.

If we shorten it to `/\.[cm]?js(\?|$)/`, then we lose the extension when we `split`, which makes it a little more challenging to handle the `.mjs` and `.cjs` extensions.

Reply:
Ah ok. Good to know 👍 The longer version should be fine then